src/CargoBuild/CargoBuildTask.cs

```diff
@@ -0,0 +1,437 @@
+        {
+            var stopwatch = new Stopwatch();
+            try
+            {
+                stopwatch.Start();
+                Log.LogMessage(MessageImportance.Normal, $"---- CargoBuild fetch Starting ----\n\n");
+                var graphLoadStopWatch = new Stopwatch();
+                graphLoadStopWatch.Start();
+
+                var evaluationContext = EvaluationContext.Create(EvaluationContext.SharingPolicy.Shared);
+
+                var graph = new ProjectGraph(
+                    [new ProjectGraphEntryPoint(project)],
+                    ProjectCollection.GlobalProjectCollection,
+                    (string projectPath, Dictionary<string, string> globalProperties, ProjectCollection projCollection) =>
+                    {
+                        var loadSettings = ProjectLoadSettings.DoNotEvaluateElementsWithFalseCondition;
+
+                        var projectOptions = new ProjectOptions
+                        {
+                            GlobalProperties = globalProperties,
+                            ToolsVersion = projCollection.DefaultToolsVersion,
+                            ProjectCollection = projCollection,
+                            LoadSettings = loadSettings,
+                            EvaluationContext = evaluationContext,
+                        };
+
+                        return ProjectInstance.FromFile(projectPath, projectOptions);
+                    });
+
+                graphLoadStopWatch.Stop();
+
+                Log.LogMessage(
+                    MessageImportance.Normal,
+                    $"CargoBuild fetch: Static graph loaded in {{0}} seconds: {{1}} nodes, {{2}} edges",
+                    Math.Round(graph.ConstructionMetrics.ConstructionTime.TotalSeconds, 3),
+                    graph.ConstructionMetrics.NodeCount,
+                    graph.ConstructionMetrics.EdgeCount);
+
+                var rustProjects = new List<string>();
+                foreach (ProjectGraphNode node in graph.ProjectNodes)
+                {
+                    bool cargoFile = File.Exists(Path.Combine(node.ProjectInstance.Directory, "cargo.toml"));
+                    if (!rustProjects.Contains(node.ProjectInstance.Directory) && cargoFile)
+                    {
+                        rustProjects.Add(node.ProjectInstance.Directory);
+                    }
+                }
+
+                var tasks = new List<Task<ExitCode>>();
+
+                Log.LogMessage(MessageImportance.Normal, $"CargoBuild, Auth Enabled: {EnableAuth}");
+
+                foreach (var projects in rustProjects)
+                {
+                    string path = projects;
+
+                    var fetchTask = RustFetchAsync(path, EnableAuth);
+                    tasks.Add(fetchTask);
+                }
+
+                await Tasks.Task.WhenAll(tasks.ToArray());
+                bool success = tasks.Select(x => (int)x.Result).Sum() == 0;
```

Comment:
This can overflow. Also, exit codes could add up to 0 despite being errors (-1 and 1).

Instead this should check if any are non-zero.